library/func_misc.py

```diff
@@ -147,3 +147,44 @@ def Call(self, pos_args, named_args):
 
         keys = [value.Str(k) for k in dictionary.keys()]  # type: List[value_t]
         return value.List(keys)
+
+
+class Len(vm._Callable):
+
+    def __init__(self):
+        # type: () -> None
+        pass
+
+    def Call(self, pos_args, named_args):
+        # type: (List[value_t], Dict[str, value_t]) -> value_t
+
+        n_args = len(pos_args)
+        if n_args != 1:
+            raise error.InvalidType("len() expects exactly one argument but %d were given" %
+                          n_args, loc.Missing)
```

Comment:
Following the conventions in `frontend/typed_args.py` for consistency. I noticed python also raises `TypeError` in this case, but it feels like it should be something else...